app/resources/api/v2/barcode_printer_resource.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Api
+  module V2
+    # Provides a JSON API representation of a barcode printer.
+    # See: http://jsonapi-resources.com/ for JSONAPI::Resource documentation
+    class BarcodePrinterResource < BaseResource
+      immutable
+
+      default_includes :uuid_object
+
+      ###
+      # Attributes
+      ###
+
+      # @!attribute [r]
+      # @return [String] The name of the barcode printer.
+      attribute :name
+
+      # @!attribute [r]
+      # @return [String] The UUID of the barcode printer.
+      attribute :uuid, readonly: true
```

Comment:
Are these readonly by default because the class is immutable?

Reply:
Yes basically.  I've been a bit torn about how to document these as I'm not sure the exported docs we're working on will tell you whether the resource is immutable or not.  We should probably find a way for it to do that.  It might be as simple as putting YARD docs above the class/module that state it.  It would need to be kept in sync by anyone updating these resources of course.